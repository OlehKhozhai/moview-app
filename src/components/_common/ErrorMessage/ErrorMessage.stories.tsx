import React from 'react';
import { Meta } from '@storybook/react';
import { text } from '@storybook/addon-knobs';

import ErrorMessage from './index';

export default {
  component: ErrorMessage,
  title: 'components/ErrorMessage',
} as Meta;

const props = () => ({
  text: text('text', 'Error message'),
});

export const ErrorMessageComponent = () => (
  <div style={{ width: 400 }}>
    <ErrorMessage {...props()} />
  </div>
);
