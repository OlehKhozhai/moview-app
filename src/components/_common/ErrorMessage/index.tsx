import React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

type ErrorMessageProps = {
  text: string;
  className?: string;
};

const ErrorMessage: React.FC<ErrorMessageProps> = ({ text, className, ...props }) => {
  return (
    <p className={cn(styles.root, className)} {...props}>
      {text}
    </p>
  );
};

export default React.memo(ErrorMessage);
