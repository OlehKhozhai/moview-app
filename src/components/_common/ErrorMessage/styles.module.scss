@import 'assets/styles/mixins/colors';

.root {
  font-size: 14px;
  color: $secondary-color;
  position: absolute;
  text-transform: uppercase;
}
