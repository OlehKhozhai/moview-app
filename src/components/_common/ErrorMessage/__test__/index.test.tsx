import React from 'react';
import { create } from 'react-test-renderer';

import ErrorMessage from '../index';

describe('ErrorMessages component', () => {
  test('should render ErrorMessage component by default', () => {
    const component = create(<ErrorMessage text="Error message" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
