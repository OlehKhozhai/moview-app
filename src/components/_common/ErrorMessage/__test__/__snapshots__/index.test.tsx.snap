// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessages component should render ErrorMessage component by default 1`] = `
<p
  className="root"
>
  Error message
</p>
`;
